tpm/constants.go

```diff
@@ -92,6 +154,47 @@ const (
 	rtTrans
 )
 
+// Locality type
+type Locality byte
+
+// Values of locality
+// Note: Localities are summable
+const (
+	locZero Locality = 1 << iota
+	locOne
+	locTwo
+	locThree
+	locFour
+)
+
+// LocaMap maps Locality values to strings for convenience
+var LocaMap = map[Locality]string{
+	locZero:  "Locality 0",
+	locOne:   "Locality 1",
+	locTwo:   "Locality 2",
+	locThree: "Locality 3",
+	locFour:  "Locality 4",
+}
+
+// PrettyPrint for Localities
+func (l Locality) String() string {
+	var retString strings.Builder
+	n, ok := LocaMap[l]
+	if !ok {
+		//In this case the given locality may a sum of localities
+		for iterator, item := range LocaMap {
+			if l&iterator != 0 {
+				retString.WriteString(item + " + ")
+			}
+		}
+		if retString.String() == "" {
+			return fmt.Sprintf("locality %d", int(l))
+		}
+		return retString.String()
+	}
+	return n
+}
```

Comment:
If only a single locality can be set at a time, you can simplify this to a switch clause instead:

```go
func (l Locality) String() string {
  switch l {
    case locZero:
      return "locality 0"
    case locOne:
      return "locality 1"
   // ...
  }
  return ""
}
```

Reply:
More than one locality can be set at a time. 
Thank you for the fast review :)

Reply:
In that case, can you get rid of
```go
n, ok := LocaMap[l]
if !ok {
  // but leave the code in here
}
```

And just have the bit within the `if !ok` clause? The code within that clause should handle all cases.

It should then look very similar to the `permission.String()` method.

Reply:
Ok :)